src/Monolog/Handler/SlackHandler.php

```diff
@@ -260,36 +151,40 @@ protected function write(array $record)
      *
      * @param  int    $level
      * @return string
+     * @deprecated Use underlying SlackRecord instead
      */
     protected function getAttachmentColor($level)
     {
-        switch (true) {
-            case $level >= Logger::ERROR:
-                return 'danger';
-            case $level >= Logger::WARNING:
-                return 'warning';
-            case $level >= Logger::INFO:
-                return 'good';
-            default:
-                return '#e3e4e6';
-        }
+        trigger_error(
+            'SlackHandler::getAttachmentColor() is deprecated. Use underlying SlackRecord instead.',
+            E_USER_DEPRECATED
+        );
+
+        return $this->slackRecord->getAttachmentColor($level);
```

Comment:
Yes. We would be breaking extensions which adhere to the Liskov principle, because we are not calling this method anymore. The deprecation notice would help, but not to people not familiar with Monolog. However, this is only about people extending `SlackHandler` as we are not changing anything for people using the public API. This is only the `protected` API. So I think we could close our eyes for that BC change here, because we have a deprecation notice for people who extended it.

@Seldaek Do you think it's OK breaking the use of classes extended `SlackHandler`?